### ISSUE TITLE
A slight issue in train_llm_from_scratch

### ISSUE THREAD

**Issue by shakewingo** [wyf3/llm_related#35]
Hi there, I found a slight issue in the script train.py in train_llm_from_scratch. Below part (around line 134) should be `mask` instead of `self.mask`, also in consideration of operating in the same device. So here's the corrected line of code:

```
            mask = torch.full((1, 1, self.config.max_seq_len, self.config.max_seq_len), float("-inf"), device=q.device)  # 初始化掩码
            mask = torch.triu(mask, diagonal=1)  # 生成上三角掩码
            scores = torch.matmul(q, k.transpose(2, 3)) / math.sqrt(self.head_dim)  # 计算注意力分数
            scores = scores + mask[:, :, :s, :s]  # 应用掩码
            scores = F.softmax(scores.float(), dim=-1).type_as(q)  # 计算 softmax
            scores = self.attention_dropout(scores)  # 应用注意力 dropout
            output = torch.matmul(scores, v)  # 计算输出
```

Please take a look when you having time. Thanks for the great video and repo BTW!

**Comment by wyf3** [comment]
感谢您的指出，已修正
